cmd/subnetcmd/upgradecmd/validate.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package upgradecmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/params"
+)
+
+var (
+	errInvalidPrecompiles      = errors.New("invalid precompiles")
+	errNoBlockTimestamp        = errors.New("no blockTimestamp value set")
+	errBlockTimestampInvalid   = errors.New("blockTimestamp is invalid")
+	errBlockTimestampInthePast = errors.New("blockTimestamp is in the past")
+	errNoPrecompiles           = errors.New("no precompiles present")
+	errEmptyPrecompile         = errors.New("the precompile has no content")
+	errNoUpcomingUpgrades      = errors.New("no valid upcoming activation timestamp found")
+)
+
+func validateUpgradeBytes(file []byte) error {
+	upgrades, err := getAllUpgrades(file)
+	if err != nil {
+		return err
+	}
+
+	allTimestamps, err := getAllTimestamps(upgrades)
+	if err != nil {
+		return err
+	}
+
+	for _, ts := range allTimestamps {
+		if time.Unix(ts, 0).Before(time.Now()) {
```

Comment:
Yes that's possible and this is a very good point. However, we currently are not supporting state of upgrades very well anyways. 

The above check should be applied only to **new** upgrades - and for new ones it is **essential**.

We need to able to track state of upgrades because upgrades which have already been applied will have to stay in the `upgrade.json`. I was planning to do this in the next iteration (but before release), but I didn't have a specific ticket yet. Will create a new ticket for this.